### ISSUE TITLE
Map keys?

### ISSUE THREAD

**Issue by kylescousin** [bithatch/snake#8]
Thank you for building this.

Is there a way to remap the keys? The lighting effects are nice, but not all that useful

Ideally I want the Tartarus to have independent key codes for its keys, at the moment they are just aliases of standard keys

**Comment by brett-smith** [comment]
Quick answer, no not currently.

Long answer ...

This is the next major focus for Snake. The next update will contain support for 3 ways of dealing with keys. There are pros and cons to all of them, and I am not entirely sure which way it will go with yet.
 
 1. The existing macro system provided by OpenRazer and partially support by the current release of Snake. This as you have found is a bit weak, and is unlikely to be developed further.
 2. This - https://github.com/openrazer/openrazer/pull/1124 - is an improvement being worked on for OpenRazer. I have found it to be better than 1), but still lacking in some areas (although the author has stated intentions to develop it further after it's first release). The next version of Snake already has support for this ready when it is released. 
 3. My own solution built into Snake. This is nearly finished, I have just got a little work to do on "Automatic Profile Switching" and it will be released.  As a teaser, here a couple of screenshots of it in action.

![image](https://user-images.githubusercontent.com/10938912/105574900-643e1a80-5d5f-11eb-9743-9e5ae06fe98a.png)

![image](https://user-images.githubusercontent.com/10938912/105575140-36f26c00-5d61-11eb-8931-3b3d18124594.png)

So in short, watch this space!

**Comment by Mowinski** [comment]
Any progress :D?
